stealth.lua

```diff
@@ -0,0 +1,107 @@
+
+cloaking.privs_to_string = minetest.privs_to_string
+function minetest.privs_to_string(privs, delim)
+    assert(type(privs) == "table")
+    delim = delim or ","
+    local list = {}
+    for priv, bool in pairs(privs) do
+        if bool and priv ~= "cloaking" then
+            list[#list + 1] = priv
+        end
+    end
+    return table.concat(list, delim)
+end
+
+
+local privs_command = minetest.registered_chatcommands["privs"].func
+minetest.override_chatcommand("privs", {
+    func = function(caller, param)
+        if minetest.check_player_privs(caller, "cloaking") or minetest.check_player_privs(caller, "admin") then
+            local f = minetest.privs_to_string
+            minetest.privs_to_string = cloaking.privs_to_string
+
+            local err, msg = privs_command(caller, param)
+
+            minetest.privs_to_string = f
+            return err, msg
+        else
+            return privs_command(caller, param)
+        end
+    end
+})
+
+minetest.override_chatcommand("mods", {
+    func = function(caller, param)
+        local modnames = minetest.get_modnames()
+
+        if not (minetest.check_player_privs(caller, "cloaking") or minetest.check_player_privs(caller, "admin")) then
+            local i
+            for idx, modname in ipairs(modnames) do
+                if modname == "cloaking" then
+                    i = idx
+                    break
+                end
+            end
+
+            if i then
+                table.remove(modnames, i)
+            end
```

Comment:
The iteration is unrequired, you can simply do:
```lua
modnames.cloaking = nil
```

Reply:
1. `modnames` is list-like, and not hash-like, so `modnames.cloaking` doesn't actually refer to anything, unless I'm really missing something.
2. I know that `a.x = nil` or `a[x] = nil` is the correct way to remove elements from hash-like tables, but is it also correct for list-like tables such as modnames? In this case it'd be something like `modnames[idx] = nil` for some appropriate idx, but i don't think that'd have the desired results?